keras_cv/layers/preprocessing/auto_contrast.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.utils import preprocessing
+
+
+class AutoContrast(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Performs the AutoContrast operation on an image.
```

Comment:
Can you add more details about the layer behavior here? eg the math part of how the pixel value is updated.

Reply:
Done